datadog/cmd/main/main.go

```diff
@@ -213,6 +213,53 @@ func (d *DatadogCostSource) getDDCostsForWindow(window opencost.Window, listPric
 	// datadog's usage API sometimes provides usages that get counted multiple times
 	// this post processing stage de-duplicates those usages and costs
 	postProcess(&ccResp)
+
+	view := "sub-org"
+	params := datadogV2.NewGetEstimatedCostByOrgOptionalParameters()
+	params.StartDate = window.Start()
+	params.EndDate = window.End()
+	params.View = &view
+	resp, r, err := d.usageApi.GetEstimatedCostByOrg(d.ddCtx, *params)
+	if err != nil {
+		log.Errorf("Error when calling `UsageMeteringApi.GetEstimatedCostByOrg`: %v\n", err)
+		log.Errorf("Full HTTP response: %v\n", r)
+		ccResp.Errors = append(ccResp.Errors, err.Error())
+	}
+
+	costs = map[string]*pb.CustomCost{}
+	for _, costResp := range resp.Data {
+		attributes := costResp.Attributes
+		for _, charge := range attributes.Charges {
+			chargeCost := float32(*charge.Cost)
+			if chargeCost == 0 {
+				continue
+			}
+
+			provId := *attributes.PublicId + "/" + *charge.ProductName
```

Comment:
Is this a new provider ID format for a particular kind of Datadog cost? If you imagine that we'll be stuck with it for 5 years, do you think its the right choice?

Reply:
I think it's the right choice, as `PublicId` plus `ProductName` will allow the provider IDs to emulate an account/cost type grouping, as represented by this cost data from Datadog.

<img width="1022" alt="Screenshot 2024-03-19 at 2 13 14 PM" src="https://github.com/opencost/opencost-plugins/assets/127428785/63ee007c-a512-4b29-83f2-1b2cd62c7019">